private/bufpkg/bufplugin/bufplugindocker/docker.go

```diff
@@ -135,12 +137,13 @@ func (d *dockerAPIClient) Build(ctx context.Context, dockerfile io.Reader, plugi
 				retErr = multierr.Append(retErr, err)
 			}
 		}()
-		// TODO: We're just logging the messages from the image build.
-		// In a future PR, we might want to add support for parsing well known messages (i.e. the image id).
-		// This would allow us to potentially skip the ImageInspectWithRaw client call below.
 		scanner := bufio.NewScanner(response.Body)
 		for scanner.Scan() {
 			d.logger.Debug(scanner.Text())
+			var message jsonmessage.JSONMessage
+			if err := json.Unmarshal([]byte(scanner.Text()), &message); err == nil && message.Error != nil {
+				return message.Error
```

Comment:
While testing, I found that both the build and push commands might return a `nil` error but then communicate an error via the streaming response body. This adds handling to return an error.